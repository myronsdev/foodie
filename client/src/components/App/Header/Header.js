import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

class Header extends Component {
  renderContent() {
    if (this.props.auth) {
      return (
        <Fragment>
          <Link className="navbar-item" to="/profile">
            Profile
          </Link>
          <a className="navbar-item" href="/api/logout">
            logout
          </a>
        </Fragment>
      )
    }
    if (!this.props.auth) {
      return (
        <Fragment>
          <a className="navbar-item" href="/login">
            login
          </a>
        </Fragment>
      )
    }
  }

  render() {
    return (
      <header className="header">
        <div className="container is-fluid">
          <nav className="navbar">
            <div className="navbar-brand">
              <Link className="navbar-item" to="/">
                Home
              </Link>
            </div>
            <div className="navbar-menu">
              <div className="navbar-end">{this.renderContent()}</div>
            </div>
          </nav>
        </div>
      </header>
    )
  }
}

Header.propTypes = {
  auth: PropTypes.string
}

export default Header
