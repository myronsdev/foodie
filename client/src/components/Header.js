import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

class Header extends Component {
  renderContent() {
    if (this.props.auth) {
      return (
        <ul>
          <li>
            <a href="/api/logout">logout</a>
          </li>
          <li>
            <Link to="/profile">Profile</Link>
          </li>
        </ul>
      )
    } else {
      return (
        <li>
          <a href="/login">login</a>
        </li>
      )
    }
  }

  render() {
    return (
      <div>
        {this.renderContent()}
        <h2>food1ed</h2>
      </div>
    )
  }
}

function mapStateToProps({ auth }) {
  return { auth }
}

export default connect(mapStateToProps)(Header)
