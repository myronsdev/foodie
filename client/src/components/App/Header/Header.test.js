import React from 'react'
import Header from './Header'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

console.log(Header)
