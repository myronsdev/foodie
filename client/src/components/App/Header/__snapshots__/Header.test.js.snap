// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="container is-fluid"
  >
    <nav
      className="navbar"
    >
      <div
        className="navbar-brand"
      >
        <a
          className="navbar-item"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
      <div
        className="navbar-menu"
      >
        <div
          className="navbar-end"
        >
          <a
            className="navbar-item"
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
          <a
            className="navbar-item"
            href="/api/logout"
          >
            logout
          </a>
        </div>
      </div>
    </nav>
  </div>
</header>
`;
